### PR TITLE
Add first draft on PMA region

### PR DESCRIPTION
**Pull request by zarubaf**
That's a first draft on the PMA user guide section. Open to suggestions/improvements.

### CODE REVIEW

File: docs/01_cva6_user/PMA.rst [hunk 1]
```diff
@@ -1,24 +1,64 @@
...
+.. _cva6_pma:
+
+PMA
+===
+
+An underlying system can have multiple physical memory attributes (PMA). CVA6
+supports three main access properties:
+
+- Non-idempotent regions (I/O regions): Regions marked as non-idempotent will
+  not be queried speculatively. Typical registers of that kind would be the UART
```
Comment (jquevremont) [inline docs/01_cva6_user/PMA.rst:28]: I guess "will not be queried speculatively" could be made more explicit for the average user.

File: docs/01_cva6_user/PMA.rst [hunk 2]
```diff
@@ -1,24 +1,64 @@
...
+
+- `ArianeCfg.NrNonIdempotentRules`: Number of active non-idempotent regions.
+   - `ArianeCfg.NonIdempotentAddrBase`: Base address of the non-idempotent region.
+   - `ArianeCfg.NonIdempotentLength`: Length of the non-idempotent region.
+- `ArianeCfg.NrExecuteRegionRules`: Number of active executable regions.
+   - `ArianeCfg.ExecuteRegionAddrBase`: Base address of the executable region.
+   - `ArianeCfg.ExecuteRegionLength`: Length of the executable region.
+- `ArianeCfg.NrCachedRegionRules`: Number of active cacheable regions.
+   - `ArianeCfg.CachedRegionAddrBase`: Base address of the cacheable region.
+   - `ArianeCfg.CachedRegionLength`: Length of the cacheable region.
```
Comment (jquevremont) [inline docs/01_cva6_user/PMA.rst:49]: Double backquotes are needed in reST for code snippets.

File: docs/01_cva6_user/PMA.rst [hunk 3]
```diff
@@ -1,24 +1,64 @@
...
+
+Unsupported PMAs
+-------
+
+Currently the following RISC-V defined PMAs are not supported:
+
+- Coherence: CVA6 does not support any coherence protocol. All memory accesses
+  are considered as non-coherent.
```
Comment (jquevremont) [inline docs/01_cva6_user/PMA.rst:57]: Suggest to add a note to mention that coherence is yet possible (OpenPiton...)

File: docs/01_cva6_user/PMA.rst [hunk 4]
```diff
@@ -1,24 +1,64 @@
...
+- Atomicity: Atomicity is delegated to the underlying system. It is up the
+  subordinate device to decide whether atomicity is supported or not.
```
Comment (jquevremont) [inline docs/01_cva6_user/PMA.rst:59]: I am not sure what subordinate device means.

**Review by jquevremont — CHANGES_REQUESTED** [review]
Minor comments to improve readability.

File: docs/01_cva6_user/PMA.rst (hunk 4 above)
Comment (zarubaf) [reply docs/01_cva6_user/PMA.rst:59]: It is the politically correct description of slave: https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/m/master-slave

Comment (jquevremont) [reply docs/01_cva6_user/PMA.rst:59]: May I suggest initiator/target as a replacement of master/slave, according to IP-XACT terminology?
cf. https://www.accellera.org/images/downloads/standards/ip-xact/IP-XACT_User_Guide_2018-02-16.pdf

Comment (zarubaf) [reply docs/01_cva6_user/PMA.rst]: Thanks, changed to initiator.

**Comment by zarubaf** [comment]
I've tried to address your comments.

**Review by jquevremont — APPROVED** [review]

**Pull request merged**